cmd/clusterctl/internal/util/objs.go

```diff
@@ -37,19 +39,32 @@ func InspectImages(objs []unstructured.Unstructured) ([]string, error) {
 
 	for i := range objs {
 		o := objs[i]
+
+		if o.GetKind() != deploymentKind && o.GetKind() != daemonSetKind {
+			continue
+		}
+
+		var podSpec v1.PodSpec
 		if o.GetKind() == deploymentKind {
 			d := &appsv1.Deployment{}
 			if err := scheme.Scheme.Convert(&o, d, nil); err != nil {
 				return nil, err
 			}
-
-			for _, c := range d.Spec.Template.Spec.Containers {
-				images = append(images, c.Image)
+			podSpec = d.Spec.Template.Spec
+		} else if o.GetKind() == daemonSetKind {
+			d := &appsv1.DaemonSet{}
+			if err := scheme.Scheme.Convert(&o, d, nil); err != nil {
+				return nil, err
 			}
+			podSpec = d.Spec.Template.Spec
+		}
```

Comment:
These `if` branches can be used to validate the above as well, if we use a switch case, the `default` case could be to `continue`